ui/src/App.js

```diff
@@ -121,10 +122,32 @@ class App extends Component {
     datasetApi.datasetGet(datasetCallback);
   }
 
-  handleChange(selectedOption) {
+  handleChange(selectedFacets) {
+    let deletedFacets = this.state.selectedFacets.filter(
+      x => !selectedFacets.includes(x)
+    );
+    deletedFacets.forEach(removed => {
+      if (this.filterMap.get(removed.label) !== undefined) {
+        this.filterMap.delete(removed.label);
+      }
+    });
+    let filterArray = this.filterMapToArray(this.filterMap);
+    this.setState({ filter: filterArray });
+
     let extraFacets = [];
-    selectedOption.forEach(option => extraFacets.push(option.value));
-    console.log(extraFacets);
+    selectedFacets.forEach(option => extraFacets.push(option.value));
+    if (extraFacets.length > 0) {
+      this.setState({ extraFacets: extraFacets });
+      this.setState({ selectedFacets: selectedFacets });
+      this.facetsApi.facetsGet(
+        { filter: filterArray, extraFacets: extraFacets },
+        this.facetsCallback
+      );
+    } else {
+      this.setState({ extraFacets: [] });
+      this.setState({ selectedFacets: [] });
+      this.facetsApi.facetsGet({ filter: filterArray }, this.facetsCallback);
```

Comment:
Side note -- I wonder if calling setState in the facetsGet callback will fix #63. I can look into that.